python/lsstdesc/pipe/task/metadetection.py

```diff
@@ -0,0 +1,112 @@
+# import lsst.daf.persistence as dafPersist  # Gen2 version
+from __future__ import annotations
+import typing
+
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+from lsst.pipe.base.struct import Struct
+import lsst.afw.table as afwTable
+from lsst.pipe.tasks.coaddBase import makeSkyInfo
+import lsst.utils
+
+
+class MetadetectConnections(pipeBase.PipelineTaskConnections,
+                            dimensions=("tract", "patch", "skymap"),
+                            # running all bands
+                            # not having instrument makes it possible to combine
+                            # calexp
+                           ):
+    calExpList = cT.Input(
+        doc="Input exposures to be resampled and optionally PSF-matched onto a SkyMap projection/patch",
+        name="calexp",
+        storageClass="ExposureF",
+        dimensions=("instrument", "visit", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    skyMap = pipeBase.connectionTypes.Input(
+        doc="Input definition of geometry/box and projection/wcs for coadded exposures",
+        name="skyMap",
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+    # This is an output catalog that is not (yet) used in this skeleton
+    catalog = cT.Output(
+        doc=("Output catalog"),
+        name='metadetectObj',
+        storageClass="DataFrame",
+        dimensions=("tract", "patch", "skymap"),
+    )
+    coadd = cT.Output(
+        doc=("Coadded image"),
+        name="ngmixCoadd",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap", "visit", "instrument")
```

Comment:
"visit" does not make sense for a coadd output dimension.  @esheldon says he wants one output per `band` so this should be `band` here.  I don't know if `skymap` and `instrument` are redundant.

Reply:
You're probably right. I think `skymap` does include `instrument` but I also see [some examples](https://github.com/lsst/pipe_tasks/blob/387f8f07a2b66205f9fa6bda9a89dcdbbef3f64c/python/lsst/pipe/tasks/makeCoaddTempExp.py#L644) in the stack where both were mentioned, so I kept them both just in case.

Reply:
With the new changes I get this error
```
Traceback (most recent call last):
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/ctrl_mpexec/21.0.0-25-g2bc75c1+976188b8f0/bin/pipetask", line 29, in <module>
    sys.exit(main())
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/ctrl_mpexec/21.0.0-25-g2bc75c1+976188b8f0/python/lsst/ctrl/mpexec/cli/pipetask.py", line 43, in main
    return cli()
  File "/software/lsstsw/stack_20210211/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/software/lsstsw/stack_20210211/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/software/lsstsw/stack_20210211/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/software/lsstsw/stack_20210211/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/software/lsstsw/stack_20210211/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/software/lsstsw/stack_20210211/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/ctrl_mpexec/21.0.0-25-g2bc75c1+976188b8f0/python/lsst/ctrl/mpexec/cli/cmd/commands.py", line 102, in run
    qgraph = script.qgraph(pipelineObj=pipeline, **kwargs)
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/ctrl_mpexec/21.0.0-25-g2bc75c1+976188b8f0/python/lsst/ctrl/mpexec/cli/script/qgraph.py", line 132, in qgraph
    qgraph = f.makeGraph(pipelineObj, args)
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/ctrl_mpexec/21.0.0-25-g2bc75c1+976188b8f0/python/lsst/ctrl/mpexec/cmdLineFwk.py", line 541, in makeGraph
    qgraph = graphBuilder.makeGraph(pipeline, collections, run, args.data_query)
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/pipe_base/21.0.0-16-g0fb55c1+b60e2d390c/python/lsst/pipe/base/graphBuilder.py", line 841, in makeGraph
    scaffolding = _PipelineScaffolding(pipeline, registry=self.registry)
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/pipe_base/21.0.0-16-g0fb55c1+b60e2d390c/python/lsst/pipe/base/graphBuilder.py", line 405, in __init__
    datasetTypes = PipelineDatasetTypes.fromPipeline(pipeline, registry=registry)
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/pipe_base/21.0.0-16-g0fb55c1+b60e2d390c/python/lsst/pipe/base/pipeline.py", line 868, in fromPipeline
    thisTask = TaskDatasetTypes.fromTaskDef(taskDef, registry=registry)
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/pipe_base/21.0.0-16-g0fb55c1+b60e2d390c/python/lsst/pipe/base/pipeline.py", line 751, in fromTaskDef
    outputs = makeDatasetTypesSet("outputs", freeze=False)
  File "/software/lsstsw/stack_20210211/stack/miniconda3-py38_4.9.2-0.4.1/Linux64/pipe_base/21.0.0-16-g0fb55c1+b60e2d390c/python/lsst/pipe/base/pipeline.py", line 742, in makeDatasetTypesSet
    raise ValueError(f"Supplied dataset type ({datasetType}) inconsistent with "
ValueError: Supplied dataset type (DatasetType('ngmixCoadd', {band, instrument, skymap, tract, patch}, ExposureF)) inconsistent with registry definition (DatasetType('ngmixCoadd', {band, instrument, skymap, physical_filter, tract, visit_s
ystem, patch, visit}, ExposureF)) for metadetect.

```